File: app/src/main/java/cm/aptoide/pt/comment/data/Comment.java
package cm.aptoide.pt.comment.data;

import java.util.Date;

public class Comment {
  private final long id;
  private final String message;
  private final User user;
  private final int replies;
  private final Date date;

  public Comment(long id, String message, User user, int replies, Date date) {
    this.id = id;
    this.message = message;
    this.user = user;
    this.replies = replies;
    this.date = date;
  }

  public Comment() {
    this.id = -1;
    this.message = "";
    this.user = null;
    this.replies = -1;
    this.date = null;
  }

  public long getId() {
    return id;
  }

  public String getMessage() {
    return message;
  }

  public User getUser() {
    return user;
  }

  public int getReplies() {
    return replies;
  }

  public Date getDate() {
    return date;
  }
}

File: app/src/main/java/cm/aptoide/pt/comments/refactor/data/Comment.kt
package cm.aptoide.pt.comments.refactor.data

import java.util.*
import kotlin.collections.ArrayList

data class Comment(val id: Long = 1, val message: String = "", val user: User? = null,
                   private val replies: ArrayList<Comment> = ArrayList(),
                   val repliesNr: Int = -1, val repliesToShowNr: Int = 0,
                   val loadingReplies: Boolean, val date: Date? = null) {

  constructor(comment: Comment, repliesToShowNr: Int) : this(comment.id, comment.message,
      comment.user, comment.replies, comment.repliesNr, repliesToShowNr, comment.loadingReplies,
      comment.date)

  constructor(comment: Comment, repliesToShowNr: Int, loadingReplies: Boolean) : this(comment.id,
      comment.message,
      comment.user, comment.replies, comment.repliesNr, repliesToShowNr, loadingReplies,
      comment.date)

  constructor(comment: Comment, loadingReplies: Boolean) : this(comment.id, comment.message,
      comment.user, comment.replies, comment.repliesNr, comment.repliesToShowNr, loadingReplies,
      comment.date)

  constructor(comment: Comment, replies: ArrayList<Comment>, loadingReplies: Boolean) : this(
      comment.id, comment.message,
      comment.user, replies, comment.repliesNr, comment.repliesToShowNr, loadingReplies,
      comment.date)


  fun setReplies(replyList: List<Comment>) {
    replies.clear()
    replies.addAll(replyList)
  }

  fun getReplies(): List<Comment> {
    return replies
  }
}